model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -251,6 +255,32 @@ def exchange_and_wait(self, dim: gtx.Dimension, *fields: gtx.Field) -> None:
         res.wait()
         log.debug(f"exchange for {len(fields)} fields of dimension ='{dim.value}' done.")
 
+    def exchange_buffers(
+        self,
+        field_dims: Sequence[gtx.Dimension],
+        *buffers: data_alloc.NDArray,
+    ) -> None:
+        """
+        Allows to exchange numpy buffers of by temporarily wrapping them into a field.
+
+        Restrictions are:
+            - buffers must be 1d or 2d
+            - all buffers must have the same shape (dimensions)
+
+        """
+        assert len(field_dims) == buffers[0].ndim, "dimensions and buffer size do not match"
+        assert buffers[0].ndim in (1, 2), "Buffers must be 1d or 2d"
+        assert (
+            field_dims[0] in dims.MAIN_HORIZONTAL_DIMENSIONS.values()
+        ), f"first dimension must be one of ({dims.MAIN_HORIZONTAL_DIMENSIONS.values()})"
+        shape = buffers[0].shape
+        for b in buffers:
+            assert b.shape == shape, "All buffers must have the same shape"
+        domain = {d: (0, shape[i]) for i, d in enumerate(field_dims)}
+        # uses gt4py internal API
+        wrapped = (gtx_common._field(b, domain=domain) for b in buffers)
```

Comment:
I don't think we need to do this if we implement this function here, see `_get_applied_pattern` where we extract the buffer from the field. My perspective:
- either consider exchanging numpy/cupy buffers a workaround, then I would recommend a free function that doesn't pollute the interface of this class, or
- make it a first class feature and implement this function directly on buffers.
Or do I miss something?

Reply:
Yes, I understand that part, but I don't see the need to wrap it in a field here, because we can directly work on the buffers.